index.d.ts

```diff
@@ -0,0 +1,52 @@
+import type { Writable } from 'node:stream';
+import type { EventEmitter } from 'node:events';
+import type { RequestHandler } from 'express';
+
+export interface IMiddlewareOptions {
+  streams?: Writable[];
+  stdout?: boolean;
+  routes?: {
+    whitelist?: string[];
+    strictChecking?: boolean;
+  };
+  userAnalytics?: {
+    platform?: boolean;
+    country?: boolean;
+  };
+  format?: {
+    type?: 'json' | 'common' | 'combined';
+    options?: {
+      cookies?: boolean;
+      headers?: boolean;
+      compact?: boolean;
+      spacing?: string | number;
+      separator?: string;
+    };
+  };
+}
+
+export interface ICustomOptions {
+  streams?: Writable[];
+  stdout?: boolean;
+  format?: {
+    type?: 'json' | 'text';
+    options?: {
+      compact?: boolean;
+      spacing?: string | number;
+      separator?: string;
+    };
+  };
+}
+
+export interface ICustomLogger {  
+  config: ICustomOptions;
+  logToConsole: boolean;
```

Comment:
Please note that the `logToConsole` property sits on the `config` level and not on the `CustomLogger` instance itself.